cpp/include/cudf/reshape.hpp

```diff
@@ -107,6 +107,29 @@ std::unique_ptr<column> byte_cast(
   rmm::cuda_stream_view stream      = cudf::get_default_stream(),
   rmm::device_async_resource_ref mr = cudf::get_current_device_resource_ref());
 
+/**
+ * @brief Copies a table into a contiguous column-major device array.
+ *
+ * This function converts a table_view with columns of the same type
+ * into a 2D device array in column-major order. The output buffer must be
+ * preallocated and large enough to hold `num_rows * num_columns` values of `output_dtype`.
+ *
+ * @throws cudf::logic_error if column types do not match `output_dtype`
+ * @throws cudf::logic_error if `output_dtype` is not fixed-width or is a fixed-point type
+ *
+ * @param input A table with fixed-width, non-nullable columns of the same type
+ * @param output Pointer to device memory large enough to hold `num_rows * num_columns` values
```

Comment:
"Large enough" is a bit unclear. We want correctly-sized allocations, not oversized allocations.
```suggestion
 * @param output Pointer to device memory sized to hold `num_rows * num_columns` values
```